lib/ex_aws/parsers.ex

```diff
@@ -0,0 +1,18 @@
+defmodule ExAws.Parsers do
+  defmacro __using__(_opts) do
+    quote do
+      import SweetXml, only: [sigil_x: 2]
+      def parse({:error, {type, http_status_code, %{body: xml}}}, _) do
+        parsed_body = xml
+        |> SweetXml.xpath(~x"//ErrorResponse",
+                          request_id: ~x"./RequestId/text()"s,
+                          type: ~x"./Error/Type/text()"s,
+                          code: ~x"./Error/Code/text()"s,
+                          message: ~x"./Error/Message/text()"s,
+                          detail: ~x"./Error/Detail/text()"s)
```

Comment:
Is this correct? Every service that uses XML has this pattern? Including S3?

I'm a little reluctant to do this because AWS services are not known for being particularly consistent with respect to one another. There is some level of consistency that exists with respect to a given Operation type like Query or JSON or RestQuery but across those there is very little.